cpp/include/cuspatial/experimental/iterator_factory.cuh

```diff
@@ -120,6 +134,26 @@ struct strided_functor {
   auto __device__ operator()(std::size_t i) { return i * stride; }
 };
 
+/**
+ * @internal
+ * @brief Functor to transform an index into a geometry ID determined by a range of offsets.
+ */
+template <class IndexT, class GeometryIter>
+struct index_to_geometry_id {
+  GeometryIter geometry_begin;
+  GeometryIter geometry_end;
+
+  index_to_geometry_id(GeometryIter begin, GeometryIter end)
+    : geometry_begin(begin), geometry_end(end)
+  {
+  }
+
+  CUSPATIAL_HOST_DEVICE auto operator()(IndexT idx)
+  {
+    return thrust::prev(thrust::upper_bound(thrust::seq, geometry_begin, geometry_end, idx));
+  }
```

Comment:
This only returns the iterator not the distance of the iterator from `geometry_begin`. And this may conflict with your example given below for `make_geometry_id_iterator`. Is this intended?

Reply:
Right... the example is wrong. Not sure what a better example would be. I think I want to return the iterator not the distance, it works in the current uses.

Reply:
Ah, I fixed using `std::distance` as you implied. And also fixed the examples to use sequentially increasing object IDs. I'm glad it works this way, since we can be sure we don't overflow the object ID given any index type.